### PR TITLE
fix: [CI-8524]: Incorporating UX and Product feedback for rollout

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
### Summary

1. Rename feature to `Harness AIDA` from `Harness AI Copilot`.
2. Show delay message only after 7 secs from api call is initiated.
3. Bug fixes identified in internal testing.

image

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/HarnessCopilot/HarnessCopilot.tsx [hunk 1]
```diff
@@ -44,6 +44,8 @@ enum AIAnalysisStatus {
   Failure = 'FAILURE'
 }
 
+const SHOW_DELAY_MSSG_AFTER_DURATION = 7000
+
 function HarnessCopilot(props: HarnessCopilotProps): React.ReactElement {
```
Comment (hemanthmantri) [inline src/modules/70-pipeline/components/HarnessCopilot/HarnessCopilot.tsx:49]: Will you change this later?

File: src/modules/70-pipeline/components/HarnessCopilot/HarnessCopilot.tsx [hunk 2]
```diff
@@ -187,7 +198,7 @@ function HarnessCopilot(props: HarnessCopilotProps): React.ReactElement {
                   }}
                   className={css.copilot}
                 >
-                  {getString('pipeline.copilot.askAICopilot')}
+                  {getString('pipeline.copilot.askAIDA')}
```
Comment (hemanthmantri) [inline src/modules/70-pipeline/components/HarnessCopilot/HarnessCopilot.tsx:201]: Keeping class name as-is?

File: src/modules/70-pipeline/components/execution/StepDetails/tabs/StepDetailsTab/StepDetailsTab.tsx [hunk 3]
```diff
@@ -54,7 +54,8 @@ export function StepDetailsTab(props: ExecutionStepDetailsTabProps): React.React
           <p>{errorMessage}</p>
         </div>
       ) : null}
-      {showHarnessCoPilot({
+      {isFailed &&
+      showHarnessCoPilot({
```
Comment (hemanthmantri) [inline src/modules/70-pipeline/components/execution/StepDetails/tabs/StepDetailsTab/StepDetailsTab.tsx:58]: Later?

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 4]
```diff
@@ -1875,14 +1875,15 @@ violationDetails: Violation details
 supplier: Supplier
 copilot:
```
Comment (hemanthmantri) [inline src/modules/70-pipeline/strings/strings.en.yaml:1876]: Need to change all of these - may be in a subsequent PR.

**Review by hemanthmantri — APPROVED** [review]

File: src/modules/70-pipeline/strings/strings.en.yaml (hunk 4 above)
Comment (vardanbansal-harness) [reply src/modules/70-pipeline/strings/strings.en.yaml:1876]: Yes, will cleanup in a follow up, to keep these hotfix changes minimal.

File: src/modules/70-pipeline/components/HarnessCopilot/HarnessCopilot.tsx (hunk 1 above)
Comment (vardanbansal-harness) [reply src/modules/70-pipeline/components/HarnessCopilot/HarnessCopilot.tsx:49]: Yes, will cleanup all copilot references.

**Review by tan-nhu — APPROVED** [review]

**Pull request merged**